code/go/0chain.net/smartcontract/storagesc/allocation_test.go

```diff
@@ -2438,8 +2434,372 @@ func TestUpdateAllocationRequest(t *testing.T) {
+
+		for i, ba := range expectedAlloc.BlobberAllocs {
+
+			blobberNode, err := ssc.getBlobber(ba.BlobberID, balances)
+			require.NoError(t, err)
+
+			// Adjust ChallengePoolIntegralValue based on price changes
+			change := changes[i]
+
+			if change.isNegative {
+				ba.ChallengePoolIntegralValue -= change.Value
+			} else {
+				ba.ChallengePoolIntegralValue += change.Value
+			}
+			// Adjust size and write price for each blobber
+			oldSize := ba.Size
+			ba.Size += uar.Size / int64(afterAllocBase.DataShards)
+			assert.Greater(t, ba.Size, oldSize, "Blobber allocation size should increase")
+
+			// get the old write price of the blobber
+			ba.Terms.WritePrice = blobberNode.mustBase().Terms.WritePrice
+		}
+
+		// Finally, compare the expected allocation with the actual one
+		compareAllocationData(t, *expectedAlloc, *afterAllocBase)
+	})
+
+	t.Run("Add blobber to unused allocation with price variations should work", func(t *testing.T) {
+		var (
+			tp     = int64(0)
+			client = newClient(2000*x10, balances)
+
+			// Allocation
+			beforeAlloc, blobbers = setupAllocationWithMockStats(t, ssc, client, tp, balances, true, false, false)
+			allocID               = beforeAlloc.ID
+		)
+		// check worker pool should be 100 zcn
+		require.Equal(t, 100*x10, int(beforeAlloc.WritePool), "Write pool should be not be zero")
+
+		// Define the number of blobbers that will increase or decrease their price
+		increasePriceCount := 5 // Number of blobbers to increase the price for
+		decreasePriceCount := 2 // Number of blobbers to decrease the price for
+
+		// Iterate over all blobbers and adjust their write price
+		for i, blobber := range blobbers {
+
+			blobberNode, err := ssc.getBlobber(blobber.id, balances)
+			require.NoError(t, err)
+
+			// Adjust price: first `increasePriceCount` blobbers increase price by 2x, next `decreasePriceCount` blobbers decrease price by 0.5x
+			if i < increasePriceCount {
+				blobberNode.mustUpdateBase(func(b *storageNodeBase) error {
+					b.Terms.WritePrice = b.Terms.WritePrice * 2
+					return nil
+				})
+			} else if i >= increasePriceCount && i < increasePriceCount+decreasePriceCount {
+				blobberNode.mustUpdateBase(func(b *storageNodeBase) error {
+					b.Terms.WritePrice = b.Terms.WritePrice / 2
+					return nil
+				})
+			}
+			_, err = updateBlobber(t, blobberNode, 0, tp, ssc, balances)
+			require.NoError(t, err)
+		}
+
+		// crdate a update allocation request
+		var uar updateAllocationRequest
+		uar.ID = allocID
+
+		// add a blobber with 3GB size
+		nb3 := addBlobber(t, ssc, 3*GB, tp, avgTerms, 50*x10, balances, false, false)
+		uar.AddBlobberId = nb3.id
+		uar.AddBlobberAuthTicket = ""
+
+		// why 5x10 - bcz while adding the blobber the alloc, we are keeping blobber size = 1 gb
+		// so tokens need to be locked to add blobber would be blobber size * write price
+		// 1 gb * 5 zcn = 5 zcn
+		resp, err := uar.callUpdateAllocReq(t, client.id, 5*x10, tp, ssc, balances)
```

Comment:
Lock bare minimum amount and also add check, less than amount should fail

Reply:
Resolved